indico/modules/payment/controllers.py

```diff
@@ -188,24 +185,16 @@ class RHPaymentEventCheckout(RHPaymentEventBase):
     """Payment/Checkout page for registrants"""
 
     def _process(self):
-        if self._registrant.getPayed():
-            flash(_('You have already paid for your registration.'), 'info')
-            return redirect(url_for('event.confRegistrationFormDisplay', self._conf, **get_registrant_params()))
-        checkout_attempt_delta = None
-        if not self._registrant.isCheckoutSessionAlive():
-            self._registrant._checkout_attempt_dt = datetime.now()
-        else:
-            checkout_attempt_dt = self._registrant.getCheckoutAttemptDt()
-            checkout_attempt_delta = int(math.ceil((datetime.now() - checkout_attempt_dt).total_seconds() / 60))
-        event = self._conf
-        amount = self._registrant.getTotal()
-        currency = event_settings.get(event, 'currency')
-        plugins = get_active_payment_plugins(event)
+        if self.registration.state.name != 'unpaid':
```

Comment:
why not use the state enum here? name checks are meant for templates where it's easier that way..